cmd/kubelet/app/server.go

```diff
@@ -380,6 +380,11 @@ func run(s *options.KubeletServer, kubeDeps *kubelet.KubeletDeps) (err error) {
 		return err
 	}
 
+	kubeClient, err := getKubeClient(s)
```

Comment:
Trade off would be that instead of plumbing the client down as done here, we would have to surface the event dispatcher by plumbing it up to here.